### ISSUE TITLE
Upsert fails on Postgres with list column.

### ISSUE THREAD

**Issue by jessehansen** [prisma/prisma#16760]
### Bug description

When upsetting a record with a list column, the query fails with `column reference \"<column_name>\" is ambiguous`.

### How to reproduce

1. Create a Postgres table with a list column.
2. Upsert to it, either pushing on update or setting on create
3. See error

### Expected behavior

Upsert should succeed

### Prisma information

Schema: I only included the users table for brevity here as I think it's the only table involved
```prisma
generator client {
    provider        = "prisma-client-js"
    previewFeatures = ["fullTextSearch"]
}

datasource db {
    provider = "postgresql"
    url      = env("WEBAPP_DATABASE_URL")
}

model User {
    id                        String               @id @default(uuid())
    givenName                 String?              @map("first_name")
    familyName                String?              @map("family_name")
    emailAddress              String?              @unique @map("email_address")
    mobileNumber              String?              @unique @map("mobile_number")
    password                  String?
    active                    Boolean              @default(true)
    contractorId              String?              @map("contractor_id")
    supplierId                String?              @map("supplier_id")
    dateCreated               DateTime             @default(now()) @map("date_created")
    dateUpdated               DateTime             @updatedAt @map("date_updated")
    registered                Boolean              @default(false)
    dateRegistered            DateTime?            @map("date_registered")
    stripeCustomerId          String?              @map("stripe_customer_id")
    mtCounterpartyId          String?              @map("mt_counterparty_id")
    roles                     RoleType[]
    //
    //
    customer                  Customer?
    contractor                Contractor?          @relation(fields: [contractorId], references: [id])
    verificationCodes         VerificationCode[]
    paymentMethods            PaymentMethod[]
    createdDisputes           Dispute[]            @relation("createdUser")
    resolvedDisputes          Dispute[]            @relation("resolvedUser")
    supplier                  Supplier?            @relation(fields: [supplierId], references: [id])
    projectAttachments        ProjectAttachment[]
    projectEvent              ProjectEvent[]
    paymentRequests           PaymentRequest[]
    requestedByRefundRequests RefundRequest[]      @relation("requestedUser")
    completedByRefundRequests RefundRequest[]      @relation("completedUser")
    changeOrders              ProjectChangeOrder[]

    @@map(name: "users")
}

enum RoleType {
    CONTRACTOR_ADMIN
    CONTRACTOR_FINANCE
    CONTRACTOR_SALES
    CUSTOMER
    PK_ADMIN
    PK_DEV
    PK_FINANCE
    SUPPLIER_ADMIN
    SUPPLIER_FINANCE

    @@map("role_type")
}
```

```ts
    await prisma.user.upsert({
      where: { emailAddress },
      update: { contractorId, roles: { push: "CONTRACTOR_SALES" } },
      create: {
        emailAddress,
        givenName,
        familyName,
        contractorId,
        roles: {
          set: "CONTRACTOR_SALES",
        },
      },
    });
```

Postgres logs:
```
2022-12-12 21:59:44.742 UTC [23418] ERROR:  column reference "roles" is ambiguous at character 273
2022-12-12 21:59:44.742 UTC [23418] STATEMENT:  INSERT INTO "public"."users" ("id","first_name","family_name","email_address","active","contractor_id","date_created","date_updated","registered","roles") VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10) ON CONFLICT ("email_address") DO UPDATE SET "contractor_id" = $11, "roles" = "roles" || $12, "date_updated" = $13 WHERE ("public"."users"."email_address" = $14 AND 1=1) RETURNING "public"."users"."id", "public"."users"."first_name", "public"."users"."family_name", "public"."users"."email_address", "public"."users"."mobile_number", "public"."users"."password", "public"."users"."active", "public"."users"."contractor_id", "public"."users"."supplier_id", "public"."users"."date_created", "public"."users"."date_updated", "public"."users"."registered", "public"."users"."date_registered", "public"."users"."stripe_customer_id", "public"."users"."mt_counterparty_id", "public"."users"."roles"
```

The issue is `"roles" = "roles" || $12` - I think it should be `"roles" = "public"."users"."roles" || $12`

### Environment & setup

- OS: MacOS, & in Vercel deployment environment
- Database: PostgreSQL
- Node.js version: v16.17.0

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 4.7.1
@prisma/client          : 4.7.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 272861e07ab64f234d3ffc4094e32bd61775599c (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 272861e07ab64f234d3ffc4094e32bd61775599c (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 272861e07ab64f234d3ffc4094e32bd61775599c (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 272861e07ab64f234d3ffc4094e32bd61775599c (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.7.1-1.272861e07ab64f234d3ffc4094e32bd61775599c
Default Engines Hash    : 272861e07ab64f234d3ffc4094e32bd61775599c
Studio                  : 0.477.0
Preview Features        : fullTextSearch
✨  Done in 1.22s.
```

**Comment by RomanHotsiy** [comment]
I have the same issue.

**Comment by janpio** [comment]
Is `updates` also an `enum` for you @RomanHotsiy?

**Comment by pimmee** [comment]
@Weakky 
I'm afraid this is not completely fixed yet. When a column has `@map("column_name")` it fails.

Here's a reproduction: https://github.com/pimmee/prisma-string-list-bug

**Comment by janpio** [comment]
This has been fixed @pimmee, but not released - it will be part of 4.11.0.

Can you try after installing the latest development version via `npm install prisma@dev @prisma/client@dev`? Then it should go away.

**Comment by pimmee** [comment]
Ah, I see. It indeed works, thanks a lot!
